sdk/core/azure-core/inc/azure/core/tracing/tracing.hpp

```diff
@@ -3,309 +3,47 @@
-       *
-       * @param exception Exception which has occurred.
-       */
-      virtual void AddEvent(std::exception const& exception) = 0;
-
-      /**
-       * @brief Set the Status of the span
+       * @brief Create a Tracer object
        *
-       * @param status Updated status of the span.
-       * @param description A description associated with the Status.
+       * @param name Name of the tracer object, typically the name of the Service client
+       * (Azure.Storage.Blobs, for example)
+       * @param version Version of the service client.
+       * @return std::shared_ptr<Azure::Core::Tracing::Tracer>
        */
-      virtual void SetStatus(SpanStatus const& status, std::string const& description = "") = 0;
-
-      /**
-       * @brief Propogate information from the current span to the HTTP request headers.
-       *
-       * @param request HTTP Request to the service. If there is an active tracing span, this will
-       * add required headers to the HTTP Request.
-       */
-      virtual void PropagateToHttpHeaders(Azure::Core::Http::Request& request) = 0;
+      virtual std::shared_ptr<Azure::Core::Tracing::_internal::Tracer> CreateTracer(
+          std::string const& name,
+          std::string const& version) const = 0;
     };
 
-    /**
-     * @brief Options used while creating a span.
-     *
-     */
-    struct CreateSpanOptions final
-    {
-      /**
-       * @brief The kind of span to be created.
-       *
-       */
-      SpanKind Kind{SpanKind::Internal};
-      /**
-       * @brief Attributes associated with the span.
-       *
-       */
-      std::unique_ptr<AttributeSet> Attributes;
+    std::shared_ptr<TracerProviderImpl> TracerImplFromTracer(
+        std::shared_ptr<TracerProvider> const&);
 
-      /**
-       * @brief Parent for the newly created span.
-       */
-      std::shared_ptr<Span> ParentSpan;
-    };
-
-    /**
-     * @brief Tracer - factory for creating span objects.
-     *
-     */
-    class Tracer {
-    public:
-      /**
-       * @brief Create new Span object.
-       *
-       * @details Creates a new span object.
-       *
-       * @note There is no concept of a "current" span, each span created is a top level span,
-       * unless the CreateSpanOptions has ParentSpan member, in which case the ParentSpan member
-       * will be the parent of the newly created span.
-       *
-       * @param spanName The name of the span to create.
-       * @param options Options to be used when creating the span.
-       * @return std::shared_ptr<Azure::Core::Tracing::Span> Newly created span.
-       */
-      virtual std::shared_ptr<Span> CreateSpan(
-          std::string const& spanName,
-          CreateSpanOptions const& options = {}) const = 0;
-
-      virtual std::unique_ptr<AttributeSet> CreateAttributeSet() const = 0;
-    };
   } // namespace _internal
 
   /**
    * @brief Trace Provider - factory for creating Tracer objects.
```

Comment:
Is this still a factory for creating tracer objects? It's a method-less class, updated doc?

Also, the customer will likely find it confusing to see a class with no methods. We should add a note in the doc to explain the intent, to help usability (and address the surprise).

Reply:
**If** we decide to reword the doc comment, you'd want to update this markdown too:
https://github.com/Azure/azure-sdk-for-cpp/blob/f7a7d3e045290724a995f8c07d5773b99138bc50/doc/DistributedTracing.md#tracing-overview

> TracerProvider - this is a factory which creates Tracer objects.

Reply:
Is what a factory for creating tracer objects?  The `__internal::TracerProviderImpl` object? If so, yes.

However customers should never interact with this object. Non-service developers cannot see this type. Service developers should be interacting with the `ServiceSpan` type. For the extremely small set of service developers who are using the raw OpenTelemetry abstractions, they really should know what they're doing, so I'm not worried about confusing them.